### DISCUSSION TITLE
i want to  pass a cmake option to my port, what's is the best practice

### DISCUSSION THREAD

**Discussion by nocanstillbb** [microsoft/vcpkg#35186] [General]
in order to  disable qt unit test, i want to  pass a option to ignore unit test when vcpkg install . The way I think of is to read the json filed  from a  json file that  put to  `vcpkg_root/port/myport`  in portfile.cmake, but I think there is a better way

**Comment by autoantwort** [comment]
If you always want to disable the test, you can use
```cmake
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        -DMY_FLAG_TO_DISABLE_TESTS=ON
)
```
If you have an option for it in the vcpkg.json file you can use
```cmake
vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
   INVERTED_FEATURES
        my-enable-test-feature          MY_FLAG_TO_DISABLE_TESTS
)

vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS 
        ${FEATURE_OPTIONS}
)
```

**Comment by nocanstillbb** [comment]
very useful to me 
thanks
